meeshkan/serve/commands.py

```diff
@@ -66,35 +67,45 @@ def _add_options(func):
         type=click.Path(exists=True, file_okay=False, resolve_path=True),
         help="Directory with callback scripts to modify behavior.",
     ),
+    click.option("-k", "--kill", is_flag=True, help="Stop the mock daemon."),
     click.option(
         "-l",
         "--log-dir",
         default=None,
         type=click.Path(exists=False, file_okay=False, resolve_path=True),
         help="Directory where server logs are written.",
     ),
+    click.option(
+        "-s", "--status", is_flag=True, help="Show the status of the mock daemon."
+    ),
 ]
 
 
-@click.group(invoke_without_command=True)
+@click.command()
```

Comment:
One idea you had that we may want to implement here, @fornwall , is to have a standardized format that combines specs and callbacks so that it can be downloaded as a package. Hard not to overengineer, but the best way I've seen that done is a toplevel `.yaml` or `.json` file that acts a sort of index and sensible defaults of where things should be located. This is the way docker works, which is also in the same broad space (service virtualization). The difference here is that we can probably have a much more opinionated layout for where things go, as docker needs to handle everything, whcih is why the `Dockerfile` has so many options.

Reply:
Hm, I currently imagine it something like

```sh
meeshkan mock --docker myimage --callbacks ... [other_options] [specs]
```

which would package the specs and the callbacks into a `myimage`. So when running that image, one gets the same mock as running the above arguments without the `--docker myimage` part, and with callbacks and specs rewritten to point at something packaged inside the docker image:

```sh
meeshkan mock --callbacks ... [other_options] [specs]
```

Reply:
To clarify: If one wants to mock to be packaged into a self-contained executable (if we end up supporting that), it would work similar to `--docker myimage` above, but with something like `--executable mymock`. When running `./mymock` it would start up similarly, with callbacks and specs used from inside the packaged mock executable.

Reply:
I think it would be great if one didn't need `--docker`, `--executable`,  `--git` etc. but the code was smart enough to figure out how to handle the input argument :)